### ISSUE TITLE
v 7.0.0 doesnt work when using highcharts lib outside bundle

### ISSUE THREAD

**Issue by pdeva** [kirjs/react-highcharts#88]
I have code like this:

```
import 'highcharts';
import ReactHighCharts = require('react-highcharts');

```

package.json:

```
dependencies: {
     "highcharts": "^4.2.3",
    "react-highcharts": "^7.0.0"
}
```

this results in following error in browser:

```
Uncaught Error: Starting with version 3 of react-highcharts, Highcharts is not bundled by default.  please include highcharts. 
```

**Comment by kirjs** [comment]
React-highcharts expects global `Highcharts` variable to be set, is that the case?

**Comment by pdeva** [comment]
not sure what you mean.
The recipe here shows that i only need to import highcharts before using react-highcharts
https://github.com/kirjs/react-highcharts/blob/master/recipes.md#using-with-highcharts-package

I am rendering in browser, not on server.

**Comment by kirjs** [comment]
what happens if you do
```javascript
import 'highcharts';
console.log(Highcharts);
import ReactHighCharts = require('react-highcharts');
```
?

**Comment by pdeva** [comment]
if i do:

```
import Highcharts = require('highcharts');
console.log(Highcharts);
import ReactHighCharts = require('react-highcharts');
```
then the highcharts object is printed out, followed by your error msg again:

```
Object {win: Window, seriesTypes: Object, LegendSymbolMixin: Object, CenteredSeriesMixin: Object, TrackerMixin: Object…}
ReactHighcharts.js:1 Uncaught Error: Starting with version 3 of react-highcharts, Highcharts is not bundled by default.  please include highcharts. 
```

if you are asking people to export an npm module for highcharts, it obviously wont be present globally.
react-highcharts.jsx needs to `require('highcharts')`

**Comment by kirjs** [comment]
Can you try this: 
```
import Highcharts = require('highcharts');
global.Highcharts = Highcharts;
import ReactHighCharts = require('react-highcharts');
```?

**Comment by pdeva** [comment]
that does work, but now it needs to be put in every single file that react-highcharts is imported.
also defeats the whole purpose of using modules in the first place.
why cant you simply `require('highcharts')` inside react-highcharts, kinda like you require 'react'.

**Comment by kirjs** [comment]
Not exporting global var is a pretty new behavior, so I didn't have a chance to handle it yet. 

requiring highcharts seems like the way to go, I'll update the lib when I have some time.

**Comment by pdeva** [comment]
i created #89 to fix this

**Comment by kirjs** [comment]
Thanks, looks like tests are failing, feel free to take a look, or leave it for me to figure out

**Comment by pdeva** [comment]
probably has something to do with the dependencies visible to your tests.
i will leave that one for you

**Comment by kirjs** [comment]
Looks like importing `highcharts` turn out to be not ideal, since the use might want to use two types of charts (e.g. ReactHighcharts and ReactHighmaps) on the same page, but they would not want to have two copies of Highcharts in their codebase. 

So I suggest passing `Highcharts` as an argument to react Highcharts. e.g.
```javascript
var Highcharts = require('highcharts');
var ReactHighcharts = require('react-highcharts')(Highcharts);
```

The code is in #90

**Comment by pdeva** [comment]
this seems more like a design issue with your code. 
I shouldnt have to pass highcharts, when you can indeed just import it.

a better design would be to change your import to:

```
require('react-highcharts/highcharts')

```

and just get rid of 'react-highcharts' as an import altogether.

**Comment by kirjs** [comment]
Should be fixed by #90, feel free to reopen if not, and thanks for contributing to design decisions :)
